src/ci/docker/host-x86_64/x86_64-gnu-gcc/Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM ubuntu:22.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  bzip2 \
+  g++ \
+  make \
+  ninja-build \
+  file \
+  curl \
+  ca-certificates \
+  python3 \
+  git \
+  cmake \
+  sudo \
+  gdb \
+  libssl-dev \
+  pkg-config \
+  xz-utils \
+  mingw-w64 \
+  zlib1g-dev \
+  libzstd-dev \
+  # libgccjit dependencies
+  flex \
+  libmpfr-dev \
+  libgmp-dev \
+  libmpc3 \
+  libmpc-dev \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY scripts/sccache.sh /scripts/
+RUN sh /scripts/sccache.sh
+
+ENV RUST_CONFIGURE_ARGS \
+ --build=x86_64-unknown-linux-gnu \
+ --enable-sanitizers \
+ --enable-profiler \
+ --enable-compiler-docs \
+ --set llvm.libzstd=true \
+ --set 'rust.codegen-backends=[\"llvm\",\"gcc\"]'
+ENV SCRIPT python3 ../x.py \
+  --stage 2 \
+  test \
+  --test-codegen-backend gcc \
+  --set 'rust.codegen-backends=[\"llvm\",\"gcc\"]'
```

Comment:
I think that the second `--set` isn't needed, because the `configure` step should produce a `bootstrap.toml` file that will already have the codegen-backends configured properly.

Reply:
Well at least it's explicitly set and in case something changes, we'll know right away when that fails.